Sources/SwiftDocC/Infrastructure/Workspace/DocumentationWorkspaceDataProvider.swift

```diff
@@ -18,6 +18,8 @@ public protocol DocumentationWorkspaceDataProvider {
     /// use `UUID().uuidString`  for the provider's identifier.
     var identifier: String { get }
 
+    var allowArbitraryCatalogDirectories: Bool { get }
```

Comment:
AFAICT this property is specific to `LocalFileSystemDataProvider`. I think it'd make be better to move it there or to a new protocol that it can conform to.